server/src/main/java/org/opensearch/search/DefaultSearchContext.java

```diff
@@ -304,6 +305,22 @@ public void preProcess(boolean rewrite) {
             }
         }
 
+        if (sliceBuilder != null && readerContext != null && readerContext instanceof PitReaderContext) {
+            int sliceLimit = indexService.getIndexSettings().getMaxSlicesPerPit();
+            int numSlices = sliceBuilder.getMax();
+            if (numSlices > sliceLimit) {
+                throw new IllegalArgumentException(
```

Comment:
Should we throw `OpenSearchRejectedException` which should result in 429 instead of 5xx

Reply:
i checked the enum, it does show 429 but that corresponds to "too many requests", should we still go ahead ?